### ISSUE TITLE
[main] date should be replaced with NDATE of prod_util

### ISSUE THREAD

**Issue by chan-hoo** [NOAA-EMC/rrfs-workflow#149]
The bash `date` call in the scripts should be replaced with the `$NDATE` call of `prod_util` for EE2 compliance.

**Comment by MatthewPyle-NOAA** [comment]
Just to clarify this issue, it really only applies to date definitions that are +/- a certain amount of time from a certain time.  This really is the job of $NDATE.    Converting some items in the RRFS ex-scripts from date to $NDATE will add numerous lines to the script (date has some slick abilities for formatting of character strings that include date/time information).

**Comment by BenjaminBlake-NOAA** [comment]
@MatthewPyle-NOAA Does $NDATE only work if the date format is YYYYMMDDHH?  That is what it seems like to me.  If that is the case, then a lot of the instances of date definitions +/- a certain amount of time from a certain time in the ex-scripts can't use $NDATE instead of date, because many of them use dates in different formats (e.g. YYYY-MM-DD, YYYYMMDD, YYYYJJJDD, etc.).  I can change the ones in YYYYMMDDHH format if that is desirable.

**Comment by MatthewPyle-NOAA** [comment]
@BenjaminBlake-NOAA Yes, pretty sure that is the case.  That was what I meant with my Feb 22 comments that it would add lines bringing the date into a format useable by $NDATE, and then more lines to pull out the date information from what $NDATE provided into the form needed for the rest of the script.

**Comment by BenjaminBlake-NOAA** [comment]
Understood, thanks!  There are a few instances where $NDATE can be used instead of date, so I'll go ahead and modify those.

**Comment by guoqing-noaa** [comment]
Sorry for my jumping into this discussion. I am curious when we can use the `date` shell command.

I met a similar situation when adding the MPAS+JEDI component into the workflow.
  
For example:
The MPAS model namelist will need the following time string:
`config_start_time = '2024-05-27_03:00:00'`

My current solution is to use `${NDATE}` for all situations where we need to find previous or next cycles and only use `date` to generate a formatted time string (treating `date` as a USH utility). For example:  
```
...
CDATElbc=$($NDATE -${offset} ${CDATE})
...
...
start_time=$(date -d "${CDATE:0:8} ${CDATE:8:2}" +%Y-%m-%d_%H:%M:%S) 
...
...
# copy output to COMOUT
CDATEp1=$($NDATE 1 ${CDATE})
timestr=$(date -d "${CDATEp1:0:8} ${CDATEp1:8:2}" +%Y-%m-%d_%H.%M.%S)
...
```
Not sure whether this is acceptable to NCO.

**Comment by BenjaminBlake-NOAA** [comment]
Hi @guoqing-noaa , thanks for asking about this.  I spoke to @MatthewPyle-NOAA and @JacobCarley-NOAA about $NDATE yesterday and whether we could simply use date instead.  @JacobCarley-NOAA plans to reach out to NCO to see what is acceptable so we will get back to you.

**Comment by JacobCarley-NOAA** [comment]
I just chatted with NCO. The unix date utility is okay to use but we need to be sure we're not using it to access system clock time owing to the fact that in the event a job needs re-running, and the system clock changes to a new hour, then the job will have different input arguments and not be reproducible. Hope this makes sense.

**Comment by BenjaminBlake-NOAA** [comment]
Thanks Jacob, I will check the workflow to see if there are any instances where the system clock time is accessed (I do not know of any).  If there aren't any I will close this issue and issue #399

**Comment by guoqing-noaa** [comment]
Thanks, @BenjaminBlake-NOAA @JacobCarley-NOAA

**Comment by MatthewPyle-NOAA** [comment]
Has come up in discussion.  Want to get fresh confirmation from NCO/Steven about the permissibility of using UNIX date.
